src/app/qgsoptions.cpp

```diff
@@ -2396,6 +2429,31 @@ void QgsOptions::saveGdalDriverList()
       }
     }
   }
+
+  // vector drivers
+  for ( int i = 0; i < lstVectorDrivers->topLevelItemCount(); i++ )
```

Comment:
you could put the existing raster loop in a lambda and then re-use it here to avoid the duplicate code

Reply:
Should be fixed now. But not sure I did everything correctly.